src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -178,6 +281,19 @@ public abstract partial class JsonNamingPolicy
     public abstract partial class JsonNode
     {
         internal JsonNode() { }
+        public abstract System.Text.Json.JsonValueKind ValueKind { get; }
+        public System.Text.Json.JsonElement AsJsonElement() { throw null; }
+        public abstract System.Text.Json.JsonNode Clone();
+        public static System.Text.Json.JsonNode DeepCopy(System.Text.Json.JsonDocument jsonDocument) { throw null; }
+        public static System.Text.Json.JsonNode DeepCopy(System.Text.Json.JsonElement jsonElement) { throw null; }
+        public static System.Text.Json.JsonNode DeepCopy(System.Text.Json.JsonNode jsonNode) { throw null; }
+        public static System.Text.Json.JsonNode GetNode(System.Text.Json.JsonElement jsonElement) { throw null; }
```

Comment:
Do changes in the node get mirrored in the element?

If not then a name like CreateNode would be better to indicate that there is a copy here. Likewise, AsJsonElement should be CreateJsonElement.

Reply:
Thank you for pointing this out! In current version changes are mirrored, so naming should be appropriate. But we can discuss what behavior do we want.